core/src/main/java/com/linecorp/armeria/common/DefaultRequestHeadersBuilder.java

```diff
@@ -173,18 +162,13 @@ public Cookies cookies() {
     @Override
     public RequestHeadersBuilder cookies(Iterable<? extends Cookie> cookies) {
         requireNonNull(cookies, "cookie");
-        final ImmutableSet.Builder<Cookie> builder = ImmutableSet.builder();
-        for (String cookieHeader : getAll(HttpHeaderNames.COOKIE)) {
-            builder.addAll(Cookie.fromCookieHeader(cookieHeader));
-        }
-        builder.addAll(cookies);
-        set(HttpHeaderNames.COOKIE, Cookie.toCookieHeader(builder.build()));
+        setters().cookie(ImmutableSet.copyOf(cookies));
```

Comment:
Doesn't this create a new set instance unnecessarily? `cookie()` could accept `Iterable<? extends Cookie>` instead?